### PR TITLE
feat: optimize manually picked peaks

### PR DESCRIPTION
**Pull request by kjappelbaum**
addresses #63

### CODE REVIEW

**Comment by lpatiny** [comment]
Should we add those properties xOptimized, yOptimized (and other variables ???). 
It also has impact on autoPeakPicking.
I think it could make sense to use those optimized values.

**Comment by kjappelbaum** [comment]
@lpatiny thinking about this, the most compatible change seems to be use the behavior we have atm in the PR given that the old default was to return "optimized" values when the optimized flag was used and to return non-optimized ones by default

File: src/util/peakPicking.ts [hunk 1]
```diff
@@ -17,23 +14,44 @@ export function peakPicking(
...
+  const result: Record<string, number> = {};
   if (optimize) {
-    targetIndex = max
-      ? xyMaxClosestYPoint({ x, y }, { target }).index
-      : xyMinClosestYPoint({ x, y }, { target }).index;
+    if (max === false) {
+      let maximumY = Math.max(...y);
+      for (let i = 0; i < y.length; i++) {
+        y[i] *= -1;
+        y[i] += maximumY; // This makes it somewhat more robust
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply src/util/peakPicking.ts:35]: i'll check upstream

File: src/util/peakPicking.ts [hunk 2]
```diff
@@ -17,23 +15,50 @@ export function peakPicking(
...
+      { x, y },
+      [{ x: x[targetIndex], y: y[targetIndex], width: expectedWidth }],
+      shapeOptions,
+    );
+
+    optimizedIndex = xFindClosestIndex(x, optimizedPeak.peaks[0].x);
+    for (let key in spectrum.variables) {
+      result[key] = spectrum.variables[key].data[optimizedIndex];
+    }
+    result.width = optimizedPeak.peaks[0].width;
```
Comment (lpatiny) [inline src/util/peakPicking.ts:56]: @kjappelbaum So we don't store the optimized values ?

For the 2 optimized values of xVariable and yVariable I would put the optimized values
```
result[xVariable] = optimizedPeak.peaks[0].x;
result[yVariable] = optimizedPeak.peaks[0].y;
```

Comment (kjappelbaum) [reply src/util/peakPicking.ts:56]: right, currently we use an x/y that is actually there and closest to the optimized position. 

```
    optimizedIndex = xFindClosestIndex(x, optimizedPeak.peaks[0].x);
    for (let key in spectrum.variables) {
      result[key] = spectrum.variables[key].data[optimizedIndex];
    }
```

but I can also use the ones the optimizer spits out, we might even make a separate flag for this.

**Pull request merged**